src/Resources/ConfigResource.php

```diff
@@ -1,52 +1,82 @@
 <?php
 namespace GravApi\Resources;
 
+use GravApi\Config\Config;
+use GravApi\Config\Constants;
+use GravApi\Models\ConfigModel;
+use GravApi\Resources\Resource;
+
 /**
  * Class ConfigResource
  * @package GravApi\Resources
  */
-class ConfigResource
+class ConfigResource extends Resource
 {
-    protected $config;
     protected $id;
 
-    public function __construct($config)
+    /**
+     * @param ConfigModel $config
+     * @return ConfigResource
+     */
+    public function __construct(ConfigModel $config)
     {
-        // we should only ever have a single item array
-        foreach ($config as $key => $value) {
-            $this->id = $key;
-        }
+        $this->id = $config->id;
+        $this->resource = $config->data;
+    }
 
-        $this->config = (object) $config[$this->id];
+    /**
+     * Returns the hypermedia array for this resource
+     *
+     * @return string
+     */
+    public function getHypermedia()
+    {
+        return [
+            'related' => $this->getRelatedHypermedia()
+        ];
     }
 
     /**
-     * Returns the plugin object as an array/json.
-     * Also accepts an array of fields by which to filter.
+     * Returns the identifier for this resource
      *
-     * @param  [array] $fields optional
-     * @return [array]
+     * @return string
      */
-    public function toJson($fields = null)
+    public function getId()
     {
-        $attributes = (array) $this->config;
+        return $this->id;
+    }
+
+    /**
+     * Returns the attributes associated with this resource
+     *
+     * @param array|null $fields
+     * @return array
+     */
+    public function getResourceAttributes()
+    {
+        $attributes = $this->resource;
 
         // Filter for requested fields
-        if ($fields) {
+        if ($this->filter) {
```

Comment:
Should this filtering even exist here? `ConfigResource` doesn't currently even support custom field filtering after refactor.

Would people even _want_ to filter out config fields from the response? I imagine only admin users would have rights to see these responses, in which case they probably want to see everything.

Therefore I think field filtering for config should be removed, same as `PluginResource`.